src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java

```diff
@@ -436,22 +431,21 @@ private void handleException(Exception e) {
             long dataStartTime = request.getStart() - delayMillis;
             long dataEndTime = request.getEnd() - delayMillis;
 
-            Optional<Exception> previousException = stateManager.fetchExceptionAndClear(adID);
+            // HC logic starts here
+            if (anomalyDetector.isMultientityDetector()) {
+                Optional<Exception> previousException = stateManager.fetchExceptionAndClear(adID);
```

Comment:
I see previous exception checking is now moved inside this if block `if (anomalyDetector.isMultientityDetector()) {`.  Does that mean the previous exception only work for HC detector?

Reply:
In the main branch, it is inside HCAD block: https://github.com/opensearch-project/anomaly-detection/blob/main/src/main/java/org/opensearch/ad/transport/AnomalyResultTransportAction.java#L477-L491

Lai wanted to pursue another design and made the original change.  But due to limited time, I followed a different design after discussing with Lai and moved the code back inside the HCAD block.

HCAD won't wait for response and return right away.  That's we check previous exceptions.  Single-stream would wait for the response and thus doesn't need to check previous exceptions.

Reply:
Got it, make sense.